FarihaIS/VanRoomies
# chore(HTTPS Client): HTTPS Client for FE/BE Communication

# Welcome to VanRoomies! 👋✈️

Fixes: #38 

## Description of the change:
This change adds adds classes for an HTTPSClientFactory, CustomTrustManager, GetHelloWorldTest (to test the communication), and Constants. 
- HTTPSClientFactory: A handy class for creating OKHTTP3 clients (based off of Denis' M1)
- CustomTrustManager: Needed for our generated certificate to be trusted (based off of Denis' M1)
- GetHelloWorldTest: a class with a function `testGetHelloWorld` used to test communication with the backend. This function is currently being called in ProfileFragment and will be removed after integration.
- Constants: A class created to contain string endpoints.

## How to manually test:
1. Start up MongoDB
2. Run `sh ../scripts/generate-cert.sh 10.0.2.2` and place the certificate/key in `backend/certs/`
3. Copy `cert.pem` into `frontend/app/src/main/res/raw/` (Note this should be untracked for security reasons)
4. Run `npm run dev`
5. Start the app
6. Check the Logcat tab and look for something like:
![image](https://github.com/FarihaIS/VanRoomies/assets/64711237/67456e56-491a-418c-80af-cadec037126b)
This should run each time we go to the Profile fragment.
